### PR TITLE
Use recommended theme from theme manager

### PR DESCRIPTION
**Pull request by timja**
See https://github.com/jenkinsci/theme-manager-plugin/pull/103

and https://github.com/jenkinsci/dark-theme-plugin/pull/194

Main thing left is adapting to system themes which may need an extension to the theme manager API, I'll have a think about that tomorrow

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,6 +62,11 @@
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>antisamy-markup-formatter</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>theme-manager</artifactId>
+      <version>1.3-rc110.51e65a_de84e0</version>
```
Comment (timja) [inline pom.xml:68]: requires https://github.com/jenkinsci/theme-manager-plugin/pull/103

Comment (timja) [reply pom.xml:68]:
```suggestion
      <version>1.3</version>
```

**Comment by timja** [comment]
@uhafner I have some follow up work for moving the UI to theme manager but this is more important I think.

**Comment by uhafner** [comment]
I doubt that I find spare time this and next week, sorry.

**Comment by timja** [comment]
> I doubt that I find spare time this and next week, sorry.

np

**Comment by uhafner** [comment]
Well, there is no unified warnings plugin for maven available right now (that takes into account new and outstanding warnings).

**Comment by timja** [comment]
> Well, there is no unified warnings plugin for maven available right now (that takes into account new and outstanding warnings).

much easier imo to either be at 0 warnings or suppress existing warnings and don't allow new ones

**Comment by uhafner** [comment]
Yes, that is what I try to do in my projects. But does CheckStyle, SpotBugs and PMD report the actual warnings in a readable way already? Then it would make sense to set the `failOnWarnings` flag to `true`.

**Comment by timja** [comment]
Yes readable just fine

File: src/main/resources/io/jenkins/plugins/prism/SourceCodeViewModel/index.jelly [hunk 2]
```diff
@@ -1,13 +1,21 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:bs="/bootstrap5">
+<j:jelly xmlns:p="/lib/prism" xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:bs="/bootstrap5">
 
   <st:header name="Content-Type" value="text/html;charset=UTF-8"/>
 
   <bs:page it="${it}" notitle="true">
 
-    <st:adjunct includes="io.jenkins.plugins.prism"/>
+    <p:prism prismConfiguration="${it.prismConfiguration}" />
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/prism/SourceCodeViewModel/index.jelly:8]: need to file a PR to warnings NG for this

Comment (timja) [reply src/main/resources/io/jenkins/plugins/prism/SourceCodeViewModel/index.jelly:8]: actually it's in this plugin, resolved in https://github.com/jenkinsci/prism-api-plugin/pull/41/commits/66dcb92a0ef09d7bb84de42c7f7c62cd965b0b2b

File: pom.xml [hunk 3]
```diff
@@ -175,7 +182,7 @@
               <goal>copy-resources</goal>
             </goals>
             <configuration>
-              <outputDirectory>${project.build.directory}/${project.artifactId}/js</outputDirectory>
+              <outputDirectory>${project.basedir}/src/main/webapp/js</outputDirectory>
```
Comment (timja) [inline pom.xml:185]: `maven-hpi-plugin` cannot find resources if they are in the build directory. They need to be in the webapp directory to be picked up.

It works on a full winstone server but not the jetty dev server.

This also breaks SNAPSHOT dependencies across different plugins, e.g. creating a SNAPSHOT of this plugin and running it in the design library did not work for me

I think there's similar issues throughout the echarts / bootstrap plugins

**Comment by timja** [comment]
Converting to draft till system theme is working

**Review by uhafner — COMMENTED** [review]
I installed the PR locally but the highlighting of the source code is gone (in the warnings plugin). Do I need to activate something to see it in action?

File: pom.xml (hunk 3 above)
Comment (timja) [reply pom.xml:185]: Unsure the impacts of changing it or if it supports multiple paths etc.

This is where the code is I believe:
https://github.com/jenkinsci/maven-hpi-plugin/blob/a041c9f3ec9675ae0ae3b56b3aa49a3c3ab7f089/src/main/java/org/jenkinsci/maven/plugins/hpi/HplMojo.java#L97-L98

> degenerates the developer experience in IntelliJ.
Without this fix the developer experience doesn't work at all in IntelliJ it's 100% broken.

I assume you use that docker dev thing here without `hpi:run`? (that's not how plugins are normally developed though)

I assume your complaint is around generated resources showing up in search, if we can't fix `maven-hpi-plugin` safely then we could likely include some config in this repo to mark it as excluded.

**Comment by timja** [comment]
> I installed the PR locally but the highlighting of the source code is gone (in the warnings plugin). Do I need to activate something to see it in action?

shouldn't but I plan to do more work before taking this out of draft as this approach doesn't work for system themes.

**Pull request closed without merging**